### ISSUE TITLE
Graceful cleanup of pubsub instance

### ISSUE THREAD

**Issue by coandco** [alisaifee/coredis#92]
If you have a coroutine loop that waits on incoming pub/sub messages, like so:
```
import asyncio
from coredis import RedisCluster
from coredis.typing import Node


async def start(rclient):
    # Make sure connections are initialized
    await rclient
    pubsub = rclient.pubsub(ignore_subscribe_messages=True)
    await pubsub.subscribe("test")
    while True:
        raw_message = await pubsub.get_message()
        if raw_message:
            print(raw_message)
        await asyncio.sleep(0.01)


if __name__ == "__main__":
    nodes = [
        {"host": "redis1", "port": 6379},
        {"host": "redis2", "port": 6379},
        {"host": "redis3", "port": 6379}
    ]
    client = RedisCluster(startup_nodes=[Node(**x) for x in nodes])
    asyncio.run(start(client))
```

With the above code, when you ctrl-C, it throws the expected KeyboardInterrupt exception, but then also complains that the pubsub object tries to do some cleanup after the event loop has already stopped:

```
# python3 test.py
^CTraceback (most recent call last):
  File "/tmp/scripts/test.py", line 24, in <module>
    asyncio.run(start(client))
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
    self._run_once()
  File "/usr/lib/python3.9/asyncio/base_events.py", line 1854, in _run_once
    event_list = self._selector.select(timeout)
  File "/usr/lib/python3.9/selectors.py", line 469, in select
    fd_event_list = self._selector.poll(timeout, max_ev)
KeyboardInterrupt
Exception ignored in: <function BasePubSub.__del__ at 0x7f0755717940>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/dist-packages/coredis/commands/pubsub.py", line 82, in __del__
  File "/usr/local/lib/python3.9/dist-packages/coredis/commands/pubsub.py", line 92, in reset
  File "/usr/local/lib/python3.9/dist-packages/coredis/pool/cluster.py", line 308, in release
  File "/usr/local/lib/python3.9/dist-packages/coredis/pool/cluster.py", line 263, in __default_node_queue
  File "/usr/lib/python3.9/asyncio/queues.py", line 35, in __init__
  File "/usr/lib/python3.9/asyncio/events.py", line 642, in get_event_loop
RuntimeError: There is no current event loop in thread 'MainThread'.
```

The only way I've found to prevent this RuntimeError is to wrap my pubsub-using code in a try/finally that calls `pubsub.close()`:
```
async def start(rclient):
    # Make sure connections are initialized
    await rclient
    pubsub = rclient.pubsub(ignore_subscribe_messages=True)
    try:
        await pubsub.subscribe("test")
        while True:
            raw_message = await pubsub.get_message()
            if raw_message:
                print(raw_message)
            await asyncio.sleep(0.01)
    finally:
        pubsub.close()
```

One possible way to address this in a more elegant way would be to give the BasePubSub object `__aenter__`/`__aexit__` methods that initialize/close it so it can be used in an `async with` block.  Thus, the code would look like this:
```
async def start(rclient):
    async with rclient.pubsub(ignore_subscribe_messages=True) as pubsub:
        await pubsub.subscribe("test")
        while True:
            raw_message = await pubsub.get_message()
            if raw_message:
                print(raw_message)
            await asyncio.sleep(0.01)
```

Either way, I think the documentation should be updated with examples that reflect proper usage.

**Comment by alisaifee** [comment]
Actually, I think you've come across a nasty bug introduced in [4.3.0](https://coredis.readthedocs.io/en/stable/release_notes.html#v4-3-0) where the blocking cluster connection pool was introduced. Specifically [this](https://github.com/alisaifee/coredis/blob/master/coredis/pool/cluster.py#L308) line where upon releasing a connection to back to the pool a new async queue is created in the call to `setdefault` **EVERYTIME** release is called.

**Comment by alisaifee** [comment]
Resolved in [4.5.3](https://github.com/alisaifee/coredis/releases/tag/4.5.3). The example you shared exits cleanly now.
